### ISSUE TITLE
Cannot convert new Datadog json fields

### ISSUE THREAD

**Issue by lturcsanyi** [laurmurclar/datadog-to-terraform#61]
**Describe the bug**
The Datadog dashboard json format has been changed, there are new fields, for example: show_title, queries, etc.

**To Reproduce**
Create a new dashboard on Datadog UI and convert it.

```json
{
   "title":"Title",
   "description":"",
   "widgets":[ 
          {
            "definition":{
               "title":"Title",
               "title_size":"16",
               "title_align":"left",
               "show_legend":true,
               "legend_layout":"auto",
               "legend_columns":[
                  "value"
               ],
               "time":{
                  
               },
               "type":"timeseries",
               "requests":[
                  {
                     "queries":[
                        {
                           "data_source":"metrics",
                           "name":"query1",
                           "query":"metric1{$account}"
                        },
                        {
                           "data_source":"metrics",
                           "name":"query2",
                           "query":"metric2{$account}"
                        }
                     ],
                     "response_format":"timeseries",
                     "style":{
                        "palette":"dog_classic",
                        "line_type":"solid",
                        "line_width":"normal"
                     },
                     "display_type":"line"
                  }
               ],
               "yaxis":{
                  "include_zero":false
               },
               "markers":[
                  
               ]
            },
            "layout":{
               "x":4,
               "y":0,
               "width":4,
               "height":2
            }
          }
       ],
   "template_variables":[
      {
         "name":"account",
         "default":"account1",
         "prefix":"account"
      }
   ],
   "layout_type":"ordered",
   "is_read_only":false,
   "notify_list":[
      
   ],
   "reflow_type":"fixed"
}
```

The error message from Chrome extension: ❗️Can't convert key 'queries'

**Comment by brianpham** [comment]
Adding onto this since it is related to the dashboard changes. 

`Can't convert key 'show_title'` when having the show_title field.
```
      "definition": {
        "title": "Kubernetes Nodes Overview",
        "type": "group",
        "background_color": "vivid_green",
        "show_title": true,
        "layout_type": "ordered",
     }
```

`❗️Can't convert key 'formulas'` when having formulas.
```
              "requests": [
                {
                  "formulas": [
                    {
                      "formula": "query1"
                    },
                    {
                      "formula": "100 - query1"
                    }
                  ],
                  "response_format": "timeseries",
                  "on_right_yaxis": false,
                  "queries": [
                    {
                      "query": "avg:system.cpu.idle{$cluster} by {host}",
                      "data_source": "metrics",
                      "name": "query1"
                    }
                  ],
                  "style": {
                    "palette": "warm",
                    "line_type": "solid",
                    "line_width": "normal"
                  },
                  "display_type": "line"
                }
              ],
```

**Comment by flavio-cadete-cko** [comment]
If you want to fork and add new properties I managed with:
```
  // new fields for query
  formulas: (v) =>
    blockList(v, "formula", (k1, v1) => convertFromDefinition(FORMULA, k1, v1)),
  response_format: (_) => "",
  queries: (v) => convertEventQuery(v)
```
```
const FORMULA = {
  formula: (v) => assignmentString("formula_expression ", v)
};

const EVENT_QUERY = {
  data_source: (v) => assignmentString("data_source", v),
  indexes: (v) => assignmentString("indexes", v),
  name: (v) => assignmentString("name", v),
  compute: (v) => block("compute", v, assignmentString),
  group_by: (v) =>
    blockList(v, "group_by", (k1, v1) => convertFromDefinition(EVENT_QUERY_GROUP_BY, k1, v1)),
  search: (v) => block("search", v, assignmentString),
};

const EVENT_QUERY_GROUP_BY = {
  facet: (v) => assignmentString("facet", v),
  limit: (v) => assignmentString("limit", v),
  sort: (v) => block("sort", v, assignmentString),
};

function convertEventQuery(value) {
  return block("query", value, (_) => blockList(value, "event_query", (k1, v1) => convertFromDefinition(EVENT_QUERY, k1, v1)));
}
```

**Comment by gene-indeed** [comment]
@riccardo-di-lorenzo Yes according to @flavio-cadete-cko's fork. 

Interestingly enough trying to convert today even flavio's fork doesn't work currently for even somewhat complicated graphs. I suspect it's something with nested json rearranging and having groups within groups ... these are sums with some counts and alias. nothing special in these graphs... no work :(
<img width="1634" alt="Screen Shot 2021-07-14 at 5 12 09 PM" src="https://user-images.githubusercontent.com/43156841/125708495-0dc24bf7-54aa-4eaa-9489-3c8259bd266f.png">

**Comment by flavio-cadete-cko** [comment]
Yeah, that's the problem of not having an official source supporting this, because each one of us has special scenarios, so it's hard to maintain for all. I suggest you doing your own fork and fill in the gaps that you need for your scenarios.

**Comment by fgm** [comment]
Just in case that's not entirely clear, there's a PR awaiting review to fix all this. https://github.com/laurmurclar/datadog-to-terraform/pull/62

**Comment by opeyemi-paystack** [comment]
Hello, please is there an update on the PR review? I'm facing this issue as well

**Comment by gene-indeed** [comment]
For everyone that's still struggling still with this. Having an app is REALLY nice but I've got myself unblocked by using terraformer https://github.com/GoogleCloudPlatform/terraformer

**Comment by dfenske** [comment]
I am also facing this issue. Why hasn't that PR gotten merged yet? Thanks.

**Comment by laurmurclar** [comment]
Sorry about the delay folks! Thanks for helping each other out. I've merged https://github.com/laurmurclar/datadog-to-terraform/pull/85 and approved https://github.com/laurmurclar/datadog-to-terraform/pull/64, which should cover the following:
- `reflow_type`
- `response_format`
- `formulas`
- `queries`
- `new_group_delay`

https://github.com/laurmurclar/datadog-to-terraform/pull/62/files has a lot more to review because of the addition of a CLI tool and other new concepts, so it'll take me longer to get through. I'm going to treat that as a separate issue, since I think a lot of the fields mentioned here are covered by the two PRs above. 

Hope the other changes help in the meantime - the new version is submitted but needs to be reviewed by the various browser app before they release it to you all. I'm going to close this issue since I think the majority of fields mentioned are covered and it's becoming too long to be useful 😄 
If there are other individual fields you'd like added, the best way to get them merged quickly is to please open PRs with just those fields and include them in `/examples`/update the tests. That makes reviewing a lot easier, since otherwise I have to do the testing myself (and who has time to write tests for other people, while also having a job and a life? 😋 )

**Comment by fgmarand** [comment]
@laurmurclar Since I had to come back to that tool for work, I eventually went all in on #62 and rewrote the whole tool in Go, dropping the browser extension altogether.

Since this is now a totally different code base in a different language, on the CLI instead of the browser, I renamed it to avoid confusion. The code now lives at https://github.com/fgm/jastify
